apps/src/templates/Button.jsx

```diff
@@ -0,0 +1,217 @@
+    base: {
+      position: 'relative',
+      height: ARROW_HEIGHT,
+      textAlign: 'left',
+      display: 'inline-block',
+    },
+    left: {
+      paddingLeft: ARROW_WIDTH,
+    },
+    right: {
+      paddingRight: ARROW_WIDTH,
+    },
+  },
+  arrowHead: {
+    base: {
+      position: 'absolute',
+
+      width: 0,
+      height: 0,
+      borderColor: 'transparent',
+      borderWidth: ARROW_WIDTH,
+      borderStyle: 'solid',
+    },
+    left: color => ({
+      left: 10 - ARROW_WIDTH,
+      borderRightColor: color,
+    }),
+    right: color => ({
+      right: 10 - ARROW_WIDTH,
+      borderLeftColor: color,
+    }),
+  },
+};
+
+style.withArrow = {
+  base: {
+    position: 'relative',
+    top: (ARROW_HEIGHT - style.base.borderWidth - 10 * 2 - 44)/2,
+    margin: 0,
+  },
+  left: {
+    paddingLeft: 0,
+  },
+  right: {
+    paddingRight: 0,
+  },
+};
+
+function buttonStyle(buttonColor, textColor=color.white) {
+  return {
+    backgroundColor: buttonColor,
+    borderColor: buttonColor,
+    color: textColor,
+  };
+}
+
+const BUTTON_TYPES = {
+  "default": {
+    style: {
+      backgroundColor: color.white,
+      borderColor: color.lightest_gray,
+      color: color.black,
+    },
+  },
+  cancel: {
+    style: buttonStyle(color.green),
+  },
+  primary: {
+    style: buttonStyle(color.orange),
+  },
+  danger: {
+    style: buttonStyle(color.red),
+  },
+  action: {
+    style: buttonStyle(color.purple),
+  },
+};
+
+const ARROW_URLS = {
+  left: '/blockly/media/tryagain-arrow-head.png',
+  right: '/blockly/media/next-arrow-head.png',
+};
+
+const BaseButton = Radium(function BaseButton(props) {
+  const sizeStyle = style[props.size || 'normal'];
+  const config = BUTTON_TYPES[props.type];
+  let styleArray = [style.base, config.style, sizeStyle];
+  return (
+    <button {...props} style={[styleArray, props.style]}>
+      {props.children}
+    </button>
+  );
+});
+BaseButton.propTypes = {
+  type: React.PropTypes.oneOf(Object.keys(BUTTON_TYPES)),
+  children: React.PropTypes.node,
+  size: React.PropTypes.oneOf(['normal', 'large']),
+};
+
+const ArrowButton = Radium(function ArrowButton(props) {
+  const config = BUTTON_TYPES[props.type];
+  if (process.env.NODE_ENV === 'development') {
+    if (!props.size === 'large') {
+      throw new Error(
+        `Arrows can only be used with large buttons. This button is "${props.size}" can't use arrows.`
+      );
+    }
+  }
+  return (
+    <div style={[style.arrow.base,
+                 style.arrow[props.arrow]]}>
+      <div style={[style.arrowHead.base,
+                   style.arrowHead[props.arrow](config.style.backgroundColor)]}/>
+      <BaseButton {...props} style={[
+        style.withArrow.base,
+        style.withArrow[props.arrow],
+      ]}/>
+    </div>
+  );
+});
+ArrowButton.propTypes = Object.assign({}, BaseButton.propTypes, {
+  arrow: React.PropTypes.oneOf(['left', 'right']).isRequired,
+});
+
+const Button = Radium(function Button(props) {
```

Comment:
Why are you using a named function here, that you immediately clobber with the Radium-ized version?